src/de.hpi.swa.trufflesqueak/src/de/hpi/swa/trufflesqueak/nodes/HandleNonLocalReturnNode.java

```diff
@@ -40,24 +38,19 @@ public static HandleNonLocalReturnNode create(final CompiledCodeObject code) {
     protected final Object doHandle(final VirtualFrame frame, final NonLocalReturn nlr,
                     @Bind final Node node,
                     @Cached final InlinedConditionProfile hasModifiedSenderProfile) {
-        aboutToReturnNode.executeAboutToReturn(frame, nlr); // handle ensure: or ifCurtailed:
+        try {
+            aboutToReturnNode.executeAboutToReturn(frame, nlr); // handle ensure: or ifCurtailed:
+        } catch (ProcessSwitch ps) {
+            LogUtils.SCHEDULING.warning("HandleNonLocalReturnNode: ProcessSwitch during AboutToReturn! ");
```

Comment:
It seems we should always have this check around `executeAboutToReturn`, so maybe we should move it into the `AboutToReturnNode`.